hypershift-operator/controllers/scheduler/azure/controllers_test.go

```diff
@@ -0,0 +1,212 @@
+						KASGoMemLimit: mustQty("2Gi"),
+					},
+				},
+				{
+					Name: "large",
+					Criteria: schedulingv1alpha1.NodeCountCriteria{
+						From: 3,
+						To:   nil,
+					},
+					Effects: &schedulingv1alpha1.Effects{
+						KASGoMemLimit: mustQty("3Gi"),
+					},
+				},
+			},
+		},
+		Status: schedulingv1alpha1.ClusterSizingConfigurationStatus{
+			Conditions: []metav1.Condition{
+				{
+					Type:   schedulingv1alpha1.ClusterSizingConfigurationValidType,
+					Status: metav1.ConditionTrue,
+				},
+			},
+		},
+	}
+
+	hostedcluster := func(mods ...func(*hyperv1.HostedCluster)) *hyperv1.HostedCluster {
+		hc := &hyperv1.HostedCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "test-namespace",
+				Name:      "test",
+				Labels: map[string]string{
+					hyperv1.HostedClusterSizeLabel: "small",
+				},
+			},
+		}
+		for _, m := range mods {
+			m(hc)
+		}
+		return hc
+	}
+
+	tests := []struct {
+		name              string
+		hc                *hyperv1.HostedCluster
+		sizingConfig      *schedulingv1alpha1.ClusterSizingConfiguration
+		expectError       bool
+		expectErrText     string
+		expectRequeue     bool
+		expectAnnotations map[string]string
+	}{
+		{
+			name:          "hosted cluster not found",
+			hc:            nil,
+			sizingConfig:  sizingConfig,
+			expectError:   false,
+			expectRequeue: false,
+		},
+		{
+			name: "hosted cluster paused",
+			hc: hostedcluster(func(hc *hyperv1.HostedCluster) {
+				hc.Spec.PausedUntil = ptr.To(time.Now().Add(time.Hour).Format(time.RFC3339Nano))
+			}),
+			sizingConfig:  sizingConfig,
+			expectError:   false,
+			expectRequeue: true,
+		},
+		{
+			name: "hosted cluster without size label",
+			hc: hostedcluster(func(hc *hyperv1.HostedCluster) {
+				delete(hc.Labels, hyperv1.HostedClusterSizeLabel)
+			}),
+			sizingConfig:  sizingConfig,
+			expectError:   false,
+			expectRequeue: false,
+		},
+		{
+			name: "invalid cluster sizing configuration",
+			hc:   hostedcluster(),
+			sizingConfig: &schedulingv1alpha1.ClusterSizingConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+				Status: schedulingv1alpha1.ClusterSizingConfigurationStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   schedulingv1alpha1.ClusterSizingConfigurationValidType,
+							Status: metav1.ConditionFalse,
+						},
+					},
+				},
+			},
+			expectError:   false,
+			expectRequeue: false,
+		},
+		{
+			name: "size configuration not found",
+			hc: hostedcluster(func(hc *hyperv1.HostedCluster) {
+				hc.Labels[hyperv1.HostedClusterSizeLabel] = "extra-large"
+			}),
+			sizingConfig:  sizingConfig,
+			expectError:   true,
+			expectErrText: "could not find size configuration for size",
+			expectRequeue: false,
+		},
+		{
+			name:          "valid hosted cluster",
+			hc:            hostedcluster(),
+			sizingConfig:  sizingConfig,
+			expectError:   false,
+			expectRequeue: false,
+			expectAnnotations: map[string]string{
```

Comment:
any reason to not validate the rest of the [effects](https://github.com/openshift/hypershift/pull/5417/files#diff-7cae82863c80900f5456f966ca5bdfc20378a56d997429c284fcf10f82d8425cR9)?

Reply:
ill add test cases for the rest of the effects to TestSetHostedClusterSchedulingAnnotations